ResumableUpload.CS.Sample/ResumeableUploadSample.cs

```diff
@@ -0,0 +1,241 @@
+                                Globals.ResumeRetries += 1;
+                                bResumeAsyncWasLaunched = true;
+                                try
+                                {
+                                   await objVideosInsertRequest.ResumeAsync(objCancellationToken);
+                                }
+                                catch
+                                {
+                                    // Error handling is done in videosInsertRequest_ProgressChanged() Event
+                                }
+                            }
+                        } while (bResumeAsyncWasLaunched && IsResumeable());
+                        objFileStream.Close();
+                    }
+                }
+            }
+        }
+        /// <summary>Determines if a .ResumeAsync should be attempted.</summary>
+        /// <returns>True then .ResumeAsync should be attempted.</returns>
+        /// <remarks>When Globals.ResumeRetries==int32.MaxValue, the upload has completed successfully and IsResumeable() will return false.</remarks>
+        private static bool IsResumeable()
+        {
+            return ((Globals.ResumeRetries < MAX_RESUME_RETRIES) && (!objCancellationToken.IsCancellationRequested));
+        }
+        /// <summary>
+        /// The event handler called by the .NET Client Library to indicate the current status of the upload.
+        /// </summary>
+        /// <param name="objUploadStatus"></param>
+        static void videosInsertRequest_ProgressChanged(Google.Apis.Upload.IUploadProgress objUploadStatus)
+        {
+            switch (objUploadStatus.Status)
+            {
+                case UploadStatus.Completed:
+                    Globals.ResumeRetries = Int32.MaxValue; // Ensure that Do..While loop terminates
+                    Console.WriteLine("Status: Completed");
+                    break;
+                case UploadStatus.Starting:
+                    Console.WriteLine("Status: Starting");
+                    break;
+                case UploadStatus.NotStarted:
+                    Console.WriteLine("Status: Not Started");
+                    break;
+                case UploadStatus.Uploading:
+                    float sngPercent = 0;
+                    Globals.ResumeRetries = 0; // Successfully transmitted something so reset resume retries counter
+                    if (Globals.FileLength > 0)
+                        {
+                        sngPercent = (float)((100.0 * objUploadStatus.BytesSent) / Globals.FileLength);
+                        }
+                    //
+                    // Display only whole percents
+                    //
+                    int p = (int)sngPercent;
+                    if (p > Globals.ProgressPercent)
+                        {
+                        string strMessage =String.Format( "Status: Uploading {0:##0}% ({1:###,###,###,###,###,###,###,###})", p, objUploadStatus.BytesSent);
```

Comment:
I believe there are more concise ways to express a large number and print it with comma separators. See
http://stackoverflow.com/questions/105770/net-string-format-to-add-commas-in-thousands-place-for-a-number
for options.